src/mvt_getter.py

```diff
@@ -0,0 +1,280 @@
+        )
+    elif vec == "line":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s from
+                        (select ST_LineMerge(way) as %s, %s from
+                            (select ST_Union(way) as %s, %s from
+                                %s
+                                where (%s)
+                                and way && ST_TileEnvelope(%s, %s, %s)
+                            group by %s
+                            ) p
+                        ) p
+          """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+            ",".join([v for v in column_map.values()]),
+        )
+    elif vec == "point":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s
+                        from %s where (%s)
+                        and way && ST_TileEnvelope(%s, %s, %s)
+                        limit 10000
+                 """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+        )
+
+    return query
+
+
+if __name__ == "__main__":
+    parser = OptionParser()
+    parser.add_option("-s", "--stylesheet", dest="filename", action="append")
+    parser.add_option("-l", "--locale", dest="locale")
+
+    (options, args) = parser.parse_args()
+
+    style = MapCSS(0, 30)
+    for style_filename in options.filename:
+        style.parse(filename=style_filename)
+
+    zooms = [
+        (1, 2),
+        (2, 3),
+        (3, 4),
+        (4, 5),
+        (5, 6),
+        (6, 7),
+        (7, 8),
+        (8, 9),
+        (9, 10),
+        (10, 11),
+        (11, 12),
+        (12, 13),
+        (13, 14),
+        (14, 30)
+    ]
+
+    for (minzoom, maxzoom) in zooms:
+        print(
+            """create or replace function public.basemap_z%s(x integer, y integer)
+        returns bytea
+        as $$
+        select (
+            (select coalesce(ST_AsMVT(tile, 'area', 4096, 'way'), '') from (%s) as tile) ||
+            (select coalesce(ST_AsMVT(tile, 'line', 4096, 'way'), '') from (%s) as tile) ||
+            (select coalesce(ST_AsMVT(tile, 'node', 4096, 'way'), '') from (%s) as tile)
+        )
+        $$
+        language sql immutable strict parallel safe;
+        """
+            % (
+                minzoom,
+                get_vectors(minzoom, maxzoom, "x", "y", style, "polygon", locale=options.locale),
+                get_vectors(minzoom, maxzoom, "x", "y", style, "line", locale=options.locale),
+                get_vectors(minzoom, maxzoom, "x", "y", style, "point", locale=options.locale),
+            )
+        )
+
+    print(
```

Comment:
if you're actually outputting the text of function and not the data, then how about calling it a "style conversion" and also making a `-r mvt` in komap.py?

Reply:
yes, I think that would make things more consistent but probably there is downside that the more files involved in my work the slower I can move forward and the more things I will likely break

Reply:
Ideally I would want to use just MapCSS parser class and keep everything mapbox and sql specific in style transformer and sql generator. So there will be core (MapCSS parser) and plugins.

Reply:
The problem I see is that you're reinventing stuff from other converters (sql mainly) that should be generalized instead.